### PR TITLE
GitHub action

### LINKED ISSUE #6: Build Github Action CI/CD
1. Linter
2. run unit test
3. build docker image

For later when we have additional steps
4. Run e2e + integration test

### PR DESCRIPTION
**Pull request by allyusd**
# Description
加上 CI/CD，包含 Dockerfile, docker-compose 及 Github Actions

等目前的 Lint, Test, Docker Build and Push 流程確認後可以再加上 Deploy

# How to test

# New Dependencies

close #6

### CODE REVIEW

**Comment by noracami** [comment]
main.yml 上需要加 `on: pull_request` 嗎

**Review by allenh — CHANGES_REQUESTED** [review]
可以把原本的 `docker-compose.yml` 刪掉這樣就會變成改名字(`docker-compose.yml` -> `docker-compose.local.yml`) 看起來比較清楚跟以後要加新的`docker-compose` 也比較方便

**Comment by allenh** [comment]
另外好奇問一下 這個有測試過可以跑嗎?

**Review by allenh — APPROVED** [review]

File: frontend/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+WORKDIR /app
+
+ENV NODE_ENV=production
+
+COPY --from=base /app/next.config.js ./
+COPY --from=base /app/public ./public
+COPY --from=base /app/.next ./.next
+COPY --from=base /app/node_modules ./node_modules
+
+CMD ["node_modules/.bin/next", "start"]
```
Comment (allenh) [inline frontend/docker/Dockerfile:25]: 這邊可以像這樣寫

`package.json`
```
{
  "scripts": {
    "start": "next start"
  }
}
```

`Dockerfile`
```
# ...
CMD ["npm", "start"]
```

File: server/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+COPY package*.json ./
+
+RUN npm ci --production
+
+COPY --from=base /app/dist/ /app/dist/
+
+EXPOSE 8000
+
+CMD ["node", "dist/main"]
```
Comment (allenh) [inline server/docker/Dockerfile:27]: 同上

**Review by allenh — CHANGES_REQUESTED** [review]

**Review by allenh — APPROVED** [review]

**Comment by allenh** [comment]
From Docker: https://github.com/nodejs/docker-node/blob/main/docs/BestPractices.md#cmd

> When creating an image, you can bypass the package.json's start command and bake it directly into the image itself. First off this reduces the number of processes running inside of your container. Secondly it causes exit signals such as SIGTERM and SIGINT to be received by the Node.js process instead of npm swallowing them.

> CMD ["node","index.js"]

Approving PR.

**Review by zhihdd — APPROVED** [review]

**Pull request merged**